### PR TITLE
[AIR] Fix `CountVectorizer` failing with big data

### PR DESCRIPTION
**Pull request by Yard1**
Signed-off-by: Antoni Baum <antoni.baum@protonmail.com>

## Why are these changes needed?

Fixes bad return type in an internal `CountVectorizer` logic that caused it to fail if a Dataset task processed multiple batches instead of just one. Test has been updated.

## Related issue number

Closes https://discuss.ray.io/t/xgboost-ray-crashes-when-used-for-multiclass-text-classification/9248/3

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/data/preprocessors/vectorizer.py [hunk 1]
```diff
@@ -248,15 +249,17 @@ def most_common(counter: Counter, n: int):
...
             tokenized = df[col].map(self.tokenization_fn).map(Counter)
             for token in sorted_tokens:
-                df[f"{col}_{token}"] = tokenized.map(lambda val: val[token])
+                series = tokenized.map(lambda val: val[token])
+                series.name = f"{col}_{token}"
+                to_concat.append(series)
 
-        # Drop original columns.
-        df.drop(columns=self.columns, inplace=True)
+        df = pd.concat(to_concat, axis=1)
```
Comment (bveeramani) [inline python/ray/data/preprocessors/vectorizer.py:262]: Is this change related to the bug?

**Review by bveeramani — APPROVED** [review]
LGTM

File: python/ray/data/tests/preprocessors/test_vectorizer.py [hunk 2]
```diff
@@ -10,26 +10,35 @@
 def test_count_vectorizer():
     """Tests basic CountVectorizer functionality."""
 
-    col_a = ["a b b c c c", "a a a a c"]
-    col_b = ["apple", "banana banana banana"]
+    # Increase data size & repartition to test for
+    # discuss.ray.io/t/xgboost-ray-crashes-when-used-for-multiclass-text-classification
+    row_multiplier = 100000
```
[the remark this answers is not in the document]
Comment (Yard1) [reply python/ray/data/tests/preprocessors/test_vectorizer.py:15]: This issue only becomes apparent at larger data sizes. This is still very small and takes just a few seconds to complete.

File: python/ray/data/preprocessors/vectorizer.py (hunk 1 above)
Comment (Yard1) [reply python/ray/data/preprocessors/vectorizer.py:262]: This is a more performant way of constructing the dataframe
